cmake/HandleEigen.cmake

```diff
@@ -46,7 +42,7 @@ else()
 
     add_library(gtsam_eigen3 INTERFACE)
 
-    target_include_directories(gtsam_eigen3 INTERFACE
+    target_include_directories(gtsam_eigen3 SYSTEM INTERFACE
```

Comment:
@dellaert @varunagrawal FYI: This "SYSTEM" prefix here allows one to raise the level of warnings in "our" code, while ignoring errors in other's "system" headers. I think it's a good thing to have.